test/test_rhat.py

```diff
@@ -36,25 +42,31 @@ def test_rhat():
     chains = [chain1, chain2, chain3, chain4]
     np.testing.assert_allclose(rhat_expected(chains), rhat(chains), atol=0.1, rtol=0.2)
 
-def test_rhat_ragged():
+
+def test_rhat_ragged() -> None:
     chain1 = [1.01, 1.05, 0.98, 0.90, 1.23]
     chain2 = [0.99, 1.00, 1.01, 1.15, 0.83, 0.95]
     chains = [chain1, chain2]
-    rhat_est = rhat(chains)
     np.testing.assert_allclose(rhat_expected(chains), rhat(chains), atol=0.1, rtol=0.2)
 
-def rhat_throws(chains):
+
+def rhat_throws(chains: List[List[float]]) -> None:
```

Comment:
For the test helper functions, the only important thing is that they have a type so mypy can perform introspection, not necessarily that they be perfectly generally to hypothetical use cases (we are the only consumer of said functions). I generally used list[list[float]] because all the existing usages have that type, though for most(all?) of them sequence[sequence[float]] would also work

Reply:
Yeah, I'm not super fussed about this, just happened to notice it. We can defer a real reusable type definition for when it's used in the main body code.